burn-compute/src/tune/base.rs

```diff
@@ -0,0 +1,71 @@
+use core::time::Duration;
+
+use spin::Mutex;
+
+use crate::server::ComputeServer;
+use crate::server::Handle;
+use crate::tune::{BenchmarkPool, Operation};
+
+use super::BenchmarkResult;
+use super::TuneBenchmark;
+
+/// Use to find and reuse the best kernel for some input
+pub struct Tuner<TB, O, S>
+where
+    TB: TuneBenchmark<O, S>,
+    O: Operation,
+    S: ComputeServer,
+{
+    benchmark_pool: Mutex<BenchmarkPool<TB, O, S>>,
```

Comment:
I think the pool should be included in the compute server, in a mutable environment, so we should probably remove the mutex and make the functions mutable!